channeldb/payments.go

```diff
@@ -105,6 +105,10 @@ const (
 	// or the final cltv delta or amount is incorrect.
 	FailureReasonPaymentDetails FailureReason = 3
 
+	// FailureReasonInsufficientBalance indicates that we didn't have enough
+	// balance to complete the payment.
+	FailureReasonInsufficientBalance = 4
```

Comment:
Do we need a db version bump? I am not completely convinced that we should add this protection against downgrading on nodes running master.

Reply:
Discussed with @Roasbeef offline. Up to now we generally don't give this level of support to users on master.